tinuous

```diff
@@ -200,7 +203,16 @@ class GHABuildLog(BuildLog):
             if run.pull_requests:
                 event_id = str(run.pull_requests[0].number)
             else:
-                event_id = "UNK"
+                # The experimental "List pull requests associated with a
+                # commit" endpoint does not return data reliably enough to be
+                # worth using, so we have to do an issue search for the
+                # matching PR instead.
+                try:
+                    event_id = client.search_issues(
+                        f"repo:{run.repository.full_name} {run.head_sha}"
+                    )[0].number
```

Comment:
and `event_id` is guaranteed to be a PR or some filtering on other fields should be done to ensure that?
When we get multiple hits -- what will be the 0th -- the latest/earliest?

Reply:
By default, the results are sorted by "best match", but I've changed the code to sort by creation time in ascending order so that the oldest PR comes first.